### PR TITLE
Sanitize secrets in the requests

### LINKED ISSUE #600: Credential info exposed when setting loglevel
When setting the TraceAll loglevel for a CSI Driver, credential info exposed.

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:

We can get the credential info from the related logs: 
```sh
$ kubectl -n openshift-manila-csi-driver logs openstack-manila-csi-controllerplugin-597b69dd74-dfc9g -c csi-provisioner
I0402 03:00:42.739045       1 request.go:1107] Response Body: {"kind":"Secret","apiVersion":"v1","metadata":{"name":"csi-manila-secrets","namespace":"openshift-manila-csi-driver","uid":"c95a3ded-f1e4-47ae-b2cd-a0a54917f8bb","resourceVersion":"7910","creationTimestamp":"2021-04-02T00:41:32Z","managedFields":[{"manager":"csi-driver-manila-operator","operation":"Update","apiVersion":"v1","time":"2021-04-02T00:41:32Z","fieldsType":"FieldsV1","fieldsV1":{"f:data":{".":{},"f:os-authURL":{},"f:os-domainName":{},"f:os-password":{},"f:os-projectID":{},"f:os-region":{},"f:os-userDomainName":{},"f:os-userName":{}},"f:type":{}}}]},"data":{"os-authURL":"aHR0c/REDACTED/92Mw==","os-domainName":"cmVkaGF0LmNvbQ==","os-password":"ZWV4M2/REDACTED/lZWc5a2U=","os-projectID":"NTQyYzZlYmQ0OGJmNDBmYTg1N2ZjMjQ1Yzc1NzJlMzA=","os-region":"cmVnaW9uT25l","os-userDomainName":"cmVkaGF0LmNvbQ==","os-userName":"cHNpLW/REDACTED/raW5z"},"type":"Opaque"}

I0402 03:00:42.740126       1 controller.go:731] CreateVolumeRequest name:"pvc-6cbd5e5a-1c63-4137-9542-cc2d96d006f7" capacity_range:<required_bytes:2147483648 > volume_capabilities:<mount:<> access_mode:<mode:SINGLE_NODE_WRITER > > parameters:<key:"type" value:"ceph" > secrets:<key:"os-authURL" value:"https:///REDACTED/.redhat.com:13000/v3" > secrets:<key:"os-domainName" value:"redhat.com" > secrets:<key:"os-password" value:"eex//REDACTED//g9ke" > secrets:<key:"os-projectID" value:"542c6ebd48bf40fa857fc245c7572e30" > secrets:<key:"os-region" value:"regionOne" > secrets:<key:"os-userDomainName" value:"redhat.com" > secrets:<key:"os-userName" value:"psi-/REDACTED/-jenkins" > 
```
**What you expected to happen**:

We should not get the credential info from the related logs.

**How to reproduce it**:
1. Install Manila CSI Driver.
2. Set the Manila CSI Driver loglevel to TraceAll
3. Create manila share with sc installed by Manila CSI Driver
oc get sc  csi-manila-ceph -o yaml
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  managedFields:
  ...
  name: csi-manila-ceph
parameters:
  csi.storage.k8s.io/node-publish-secret-name: csi-manila-secrets
  csi.storage.k8s.io/node-publish-secret-namespace: openshift-manila-csi-driver
  csi.storage.k8s.io/node-stage-secret-name: csi-manila-secrets
  csi.storage.k8s.io/node-stage-secret-namespace: openshift-manila-csi-driver
  csi.storage.k8s.io/provisioner-secret-name: csi-manila-secrets
  csi.storage.k8s.io/provisioner-secret-namespace: openshift-manila-csi-driver
  type: ceph
provisioner: manila.csi.openstack.org
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

4. Check the driver log

### PR DESCRIPTION
**Pull request by Fedosin**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR uses the common lib to strip credentials in the logs.

**Which issue(s) this PR fixes**:
Fixes #600

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -730,7 +731,7 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 	createCtx := markAsMigrated(ctx, result.migratedVolume)
 	createCtx, cancel := context.WithTimeout(createCtx, p.timeout)
 	defer cancel()
-	klog.V(5).Infof("CreateVolumeRequest %+v", req)
+	klog.V(5).Infof("CreateVolumeRequest %+v", protosanitizer.StripSecrets(req))
```
Comment (msau42) [inline pkg/controller/controller.go:734]: I think we could even remove these logs entirely. It should already be logged properly via https://github.com/kubernetes-csi/csi-lib-utils/blob/44b91e2a7b0af8199fc65c1142f664fd5f7abb96/connection/connection.go#L112

@chrishenzie can you followup? I think we'll also want to backport the fixes

Comment (chrishenzie) [reply pkg/controller/controller.go:734]: Fix here:
https://github.com/kubernetes-csi/external-provisioner/pull/604

**Pull request merged**